### PR TITLE
Update state transition and types to v0.8.1

### PR DESCRIPTION
**Pull request by wemeetagain**
Passing v0.8.1 spec tests (when run file-by-file)

### CODE REVIEW

**Comment by GregTheGreek** [comment]
Something peculiar happened, it looks like a bunch of dot files got renamed and the dot is now a suffix not a prefix
example: `.bablerc` is now `babelrc.`

File: lerna.json [hunk 1]
```diff
@@ -2,5 +2,6 @@
   "packages": ["packages/*"],
   "npmClient": "yarn",
   "useWorkspaces": true,
-  "version": "independent"
+  "version": "independent",
+  "stream": "true"
```
Comment (GregTheGreek) [inline lerna.json:6]: I can't seem to find documentation, what does this do?

File: packages/lodestar/src/util/createValue.ts [hunk 2]
```diff
@@ -0,0 +1,63 @@
+import BN from "bn.js";
+import assert from "assert";
+import {AnySSZType, FullSSZType, Type, parseType, assertValidValue} from "@chainsafe/ssz";
+import {BitList, BitVector} from "@chainsafe/bit-utils";
+
+import {intDiv} from "./math";
+
+// eslint-disable-next-line @typescript-eslint/no-explicit-any
+export function createValue(type: AnySSZType, defaultValue: any = null): any {
```
Comment (GregTheGreek) [inline packages/lodestar/src/util/createValue.ts:14]: Can we get a description of what _createValue does?

File: packages/lodestar/src/chain/stateTransition/block/operations/transfer.ts [hunk 3]
```diff
@@ -35,29 +35,29 @@ export function processTransfer(
   state: BeaconState,
   transfer: Transfer
 ): void {
-  // Verify the amount and fee aren't individually too big (for anti-overflow purposes)
+  // Verify the balance the covers amount and fee
   const senderBalance = state.balances[transfer.sender];
-  assert(senderBalance.gte(transfer.amount));
-  assert(senderBalance.gte(transfer.fee));
+  assert(senderBalance.gte(transfer.amount.add(transfer.fee)));
```
Comment (GregTheGreek) [inline packages/lodestar/src/chain/stateTransition/block/operations/transfer.ts:40]: From the spec 
```
    assert state.balances[transfer.sender] >= max(transfer.amount + transfer.fee, transfer.amount, transfer.fee)
```
I believe this should:
```
assert(senderBalance.gte(BNMax(transfer.amount.add(transfer.fee), transfer.amount, transfer.fee)))
```
This being said I would like to think `BNMax` might need to be changed to handle an arbitrarily large amount of arguments.

**Comment by wemeetagain** [comment]
@GregTheGreek  can you check again? I saw the same thing earlier but now it looks like its gone away...

**Review by GregTheGreek — CHANGES_REQUESTED** [review]
I think I caught one thing

I'll give this another pass tomorrow, but most of the changes were fairly trivial less a few math ones

File: packages/lodestar/src/chain/stateTransition/block/operations/transfer.ts (hunk 3 above)
Comment (wemeetagain) [reply packages/lodestar/src/chain/stateTransition/block/operations/transfer.ts:40]: Not entirely sure why the spec does it that way.
`transfer.fee` and `transfer.amount` are both uints, so they will not be negative.
`transfer.fee + transfer.amount` will always be the biggest.

Comment (GregTheGreek) [reply packages/lodestar/src/chain/stateTransition/block/operations/transfer.ts:40]: its an overflow check - I think thats teh only reason why

File: lerna.json (hunk 1 above)
Comment (mpetrunic) [reply lerna.json:6]: It streams output of subcommands allowing you to preserve colors in output

**Review by mpetrunic — COMMENTED** [review]
could you add `--max-old-space-size=7168` to mocha when running spec tests. It should enable loading all test files into memory so we don't have failed CI.

**Comment by mpetrunic** [comment]
> could you add `--max-old-space-size=7168` to mocha when running spec tests. It should enable loading all test files into memory so we don't have failed CI.

@wemeetagain just merge 0.8 branch here. I've added flag there and it's working in travis

**Comment by GregTheGreek** [comment]
@wemeetagain it seems that the dot issue is fixed, that was weird /shrug

File: packages/lodestar/src/chain/stateTransition/block/operations/transfer.ts (hunk 3 above)
Comment (wemeetagain) [reply packages/lodestar/src/chain/stateTransition/block/operations/transfer.ts:40]: Ok, I don't think we need it in that case.

File: packages/lodestar/src/util/createValue.ts (hunk 2 above)
Comment (wemeetagain) [reply packages/lodestar/src/util/createValue.ts:14]: Added documentation to `createValue`

**Review by GregTheGreek — APPROVED** [review]
Did a re-review everything LGTM, let's ship this and get light client started 🚀

**Comment by wemeetagain** [comment]
passing tests ~~~ ready for re-review

**Comment by GregTheGreek** [comment]
@wemeetagain wait how? I thought the CI would die

**Comment by wemeetagain** [comment]
@mpetrunic  added a `--max-old-space-size=7168` to mocha

**Comment by mpetrunic** [comment]
Flag increases node heap allocation and since travis VMs come with 7.5Gb of ram, we set heap to be 7Gb

**Comment by mpetrunic** [comment]
Its anoying because you need 16gb of ram locally to do it 😕

**Review by mpetrunic — APPROVED** [review]

**Comment by GregTheGreek** [comment]
oh good god...

**Pull request merged**